wandb/integration/keras/callbacks/metrics_logger.py

```diff
@@ -50,24 +50,34 @@ def __init__(
         else:
             self.log_freq = log_freq
         self.global_batch = 0
+        self.global_step = 0
 
         if isinstance(log_freq, int):
             # define custom x-axis for batch logging.
             wandb.define_metric("batch/batch_step")
             # set all batch metrics to be logged against batch_step.
             wandb.define_metric("batch/*", step_metric="batch/batch_step")
 
+    def _get_lr(self):
+        try:
+            return self.model.optimizer.learning_rate.numpy().item()
+        except AttributeError:
+            return self.model.optimizer.learning_rate(step=self.global_step).numpy()
+
     def on_epoch_end(self, epoch: int, logs: Optional[Dict[str, Any]] = None) -> None:
         """Called at the end of an epoch."""
         wandb.log({"epoch": epoch}, commit=False)
+        logs["learning_rate"] = self._get_lr()
```

Comment:
put this inside an if-else block. When `isinstance(self.log_freq, int)`, we should not log the `learning_rate` here since it will be logged by `on_batch_end`